storage/integration_test.go

```diff
@@ -2799,197 +2797,158 @@ func TestIntegration_RequesterPays(t *testing.T) {
+			// The storage service may perform validation in any order (perhaps in parallel),
+			// so if we delete or update an object that doesn't exist and for which we lack permission,
+			// we could see either of those two errors. (See Google-internal bug 78341001.)
+			objectName := "acl-go-test" + uidSpace.New()
+			h.mustWrite(requesterPaysBucket.Object(objectName).NewWriter(ctx), []byte("hello"))
+
+			// Set up the bucket to use depending on the test case
+			bucket := test.client.Bucket(bucketName)
+			if test.userProject != nil {
+				bucket = bucket.UserProject(*test.userProject)
+			}
+
+			// Get bucket attrs
+			attrs, err := bucket.Attrs(ctx)
+			checkforErrors("get bucket attrs", err)
+			if attrs != nil {
+				if got, want := attrs.RequesterPays, true; got != want {
+					t.Fatalf("attr.RequesterPays = %t, want %t", got, want)
+				}
+			}
+
+			// Object operations
+			checkforErrors("write object", writeObject(ctx, bucket.Object(objectName), "text/plain", []byte("hello")))
+			_, err = readObject(ctx, bucket.Object(objectName))
+			checkforErrors("read object", err)
+			_, err = bucket.Object(objectName).Attrs(ctx)
+			checkforErrors("get object attrs", err)
+			_, err = bucket.Object(objectName).Update(ctx, ObjectAttrsToUpdate{ContentLanguage: "en"})
+			checkforErrors("update object", err)
+
+			// Bucket ACL operations
+			// Create another object for these to avoid object rate limits.
+			objectName2 := "acl-go-test" + uidSpace.New()
+			err = writeObject(ctx, bucket.Object(objectName2), "text/plain", []byte("hello"))
+			if err == nil {
+				// only delete the object if properly written to
+				defer h.mustDeleteObject(bucket.Object(objectName2))
+			}
+			checkforErrors("write object 2", err)
+
+			entity := ACLEntity("domain-google.com")
+
+			checkforErrors("bucket acl set", bucket.ACL().Set(ctx, entity, RoleReader))
+			_, err = bucket.ACL().List(ctx)
+			checkforErrors("bucket acl list", err)
+			checkforErrors("bucket acl delete", bucket.ACL().Delete(ctx, entity))
+
+			// Default object ACL operations
+			checkforErrors("default object acl set", bucket.DefaultObjectACL().Set(ctx, entity, RoleReader))
+			_, err = bucket.DefaultObjectACL().List(ctx)
+			checkforErrors("default object acl list", err)
+			checkforErrors("default object acl delete", bucket.DefaultObjectACL().Delete(ctx, entity))
+
+			// Object ACL operations
+			checkforErrors("object acl set", bucket.Object(objectName2).ACL().Set(ctx, entity, RoleReader))
+			_, err = bucket.Object(objectName2).ACL().List(ctx)
+			checkforErrors("object acl list", err)
+			checkforErrors("object acl list", bucket.Object(objectName2).ACL().Delete(ctx, entity))
+
+			// Copy and compose
+			_, err = bucket.Object("copy").CopierFrom(bucket.Object(objectName)).Run(ctx)
+			checkforErrors("copy", err)
+			_, err = bucket.Object("compose").ComposerFrom(bucket.Object(objectName), bucket.Object("copy")).Run(ctx)
+			checkforErrors("compose", err)
```

Comment:
It seems a bit overkill to check so many operations, but I guess that's what the test does currently?

Reply:
Yes, that is what the current test does.